### ISSUE TITLE
Code is unreachable

### ISSUE THREAD

**Issue by Groni3000** [microsoft/pylance-release#5646]
## Environment data

-   Language Server version: Pylance async client (2024.3.1) started with python extension (2024.2.1)
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.5 via python -m venv

## Code Snippet

```python
import pandas as pd
from typing import Any, Never


def test_func():
    series_1: Any = pd.Series()
    series_2: Any = pd.Series()
    test = (pd.concat([series_1, series_2], axis=1) == True).all(
        axis=1
    )  # ? `test` type is Never?

    just_some_code = 10 + 13  # ? Code is unreachable?


def one_more_test_func():
    x: Any = 10
    y: Any = 10
    test: Never  # ? Let's reproduce that `Never` type

    just_some_code = x + y  # ? It's ok...

```

## Repro Steps

1. Install venv.
2. Install pandas.
3. Use code snippet.

**Important**
1. I forced `series_1` and `series_2` to be of type `Any` because in my real code their calculation is to complex to infer type, but it's pd.Series for sure. If you change `series_1` and|or `series_2` types to be `pd.Series` - code is reachable again.
2. Seems like `Never` type of `test` variable is not the root of the problem as shown in second test function. 
3. You can delete `.all(axis=1)` and bug remains. I left it to make sure it's close enough to my real code. Still... Can delete.

## Expected behavior

Code should be reachable.

## Actual behavior

Code is unreachable.

## Logs

```
Sorry, it's too much for github. I've already lost my bug description once. So I can't paste log output here...
```

**Comment by Groni3000** [comment]
Rolling back to previous version of VSCode extension resolves the bug btw. Forgot to mention it.

**Comment by Groni3000** [comment]
Seems like I did it in a wrong place, sorry.
I looked a little bit more in issues and found this one: [#5643](https://github.com/microsoft/pylance-release/issues/5643).
Following link in that issue I found this issue is related to pandas-stubs and pylance is correct. My second example with `Never` is not working as "intendent" because I didn't initialize `Never` object... For example:

```python
from typing import Any, Never


def f() -> Never: ...

def one_more_test_func():
    x: Any = 10
    y: Any = 10
    test = f()  # ? Let's reproduce that `Never` type

    just_some_code = x + y  # ? Code is unreachable as intended.
```

If I did it earlier, I wouldn't open this issue. Sorry again.
